Https fix

aapomm opened the pull request:
[Trello](https://trello.com/c/lRuQbBBZ/274-3-when-use-the-burp-suite-i-want-to-send-an-issue-to-the-dradis-server-through-https-so-i-can-seamlessly-use-burp-with-dradis)

- Replaced ruby's `Net::HTTP` with Burp API

aapomm committed:
Replace ruby net http with burp api

aapomm committed:
Add Dradis CE support and refactor http request code

aapomm committed:
Remove unused checkbox

dormi commented:
we may bump the version to 0.0.3

dormi commented on burp-dradis.rb:
```diff
@@ -447,48 +436,68 @@ def create_dradis_issue(issue)
+      host = endpoint.sub(/^https?\:\/\//, '')
+      url = java.net.URL.new(path)
 
+      # Build the initial request
```
I reimagined this part like this:

```ruby
# Build initial request
req_as_bytes  = @helpers.toggleRequestMethod(@helpers.build_http_request(url))
req_as_string = String.from_java_bytes(req_as_bytes)
 
# Edit the request headers
req_as_string.sub! "Content-Length: 0", "Content-Length: #{payload.length}"
req_as_string.sub! "Content-Type: application/x-www-form-urlencoded", "Content-Type: application/json"
req_as_string.sub! "application/json", "application/json\n#{auth_headers}" #FIXME: this seems unclear to me

# Add body
req_as_string << payload

# Send request
response = @callbacks.make_http_request(host, port, use_ssl, req_as_string.to_java_bytes)
```
I  was trying to:
- avoid overwriting `report_req` variable, I got confused with that
- make this line more clear: `request_str.sub!('x-www-form-urlencoded', "json\n#{auth_header}")` Didin't achieve that though. Ideas welcome. Found this link: https://github.com/codewatchorg/bypasswaf/blob/master/bypasswaf.py

aapomm replied:
Using `build_http_message` as opposed to `build_http_request` gives me the ability to do the header adding but the initial http method, url, and host is omitted from the http message. This means I'll have to the ugly concatenation regardless.

aapomm committed:
Bump version to 0.0.3

aapomm committed:
Improve code readability

dormi commented:
Tested your changes @aapomm, work great.

---

One more question: 

In burp suite, we are giving the option to use Dradis CE. But I'm not sure if token authentication would work in CE. I guess not.
Should we remove the radio from Burp, or use add basic http authentication?
<img width="464" alt="captura de pantalla 2017-02-06 a las 12 23 52" src="https://cloud.githubusercontent.com/assets/85766/22645532/c767fc06-ec67-11e6-938a-861c706c91bc.png">

aapomm commented:
@dormi It _should_ work for Dradis CE if you use the shared password as the API Token.

dormi commented:
Indeed it works! 
I was using port 3000, and the port is hardcoded to 80.

I think that making ports configurable should be a separated feature.

etdsoft committed:
Build the HTTP request from scratch

Instead of using Burp's API to build it and then replace / substrings of
the result.

etdsoft commented:
I think we're going down a slippery slope here by using Burp to build a request and then calling `gsub` to replace the bits and pieces that we don't want to use.

I implemented an alternative approach in which we just build the HTTP request message from scratch here:

https://github.com/dradis/burp-dradis/pull/2

I've also broken down the main method into two: 1) build the request; 2) send the request.

Let me know what you think.

etdsoft committed:
Tweak docs/comments

etdsoft committed:
Merge pull request #2 from dradis/build-request

Build the HTTP request from scratch
